### PR TITLE
Fix rabbitmq exchange declare

### PR DESCRIPTION
**Pull request by SpikeWong**
# Description

Fix rabbitmq exchange declare misconfiguration

## Issue reference

#2764 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by berndverst** [comment]
Please create a new test case for this PR here: https://github.com/dapr/components-contrib/tree/master/tests/certification/pubsub/rabbitmq

And update unit tests

File: pubsub/rabbitmq/rabbitmq.go [hunk 1]
```diff
@@ -358,7 +358,7 @@ func (r *rabbitMQ) prepareSubscription(channel rabbitMQChannelBroker, req pubsub
 		// declare dead letter exchange
 		dlxName := fmt.Sprintf(defaultDeadLetterExchangeFormat, queueName)
 		dlqName := fmt.Sprintf(defaultDeadLetterQueueFormat, queueName)
-		err = r.ensureExchangeDeclared(channel, dlxName, fanoutExchangeKind)
+		err = r.ensureExchangeDeclared(channel, dlxName, fanoutExchangeKind, true, r.metadata.DeleteWhenUnused)
```
Comment (berndverst) [inline pubsub/rabbitmq/rabbitmq.go:361]: why is durable always true in this case? would users of the deadletter feature expect this to be the case?

Comment (SpikeWong) [reply pubsub/rabbitmq/rabbitmq.go:361]: I am following the comment https://github.com/dapr/components-contrib/pull/2836/files#diff-6d69b213107781018930a58cb30f686338377af29091816676b2e22986bb84acR369, so should I add configuration field to allow user declare this?

Comment (berndverst) [reply pubsub/rabbitmq/rabbitmq.go:361]: Got it.

**Review by berndverst — APPROVED** [review]

**Pull request merged**